app/utils/transformer/LocationOfGoodsTransformer.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils.transformer
+
+import cats.data.OptionT
+import config.Constants.QualifierOfTheIdentification.{AuthorisationNumberIdentifier, CustomsOfficeIdentifier, EoriNumberIdentifier}
+import generated.LocationOfGoodsType04
+import models.{RichAddressType06, RichCC015CType, RichGNSSType, UserAnswers}
+import pages.locationOfGoods.*
+import pages.locationOfGoods.contact.{NamePage, PhoneNumberPage}
+import services.{CountriesService, CustomsOfficesService, LocationOfGoodsIdentificationTypeService, LocationTypeService}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class LocationOfGoodsTransformer @Inject() (
+  service: LocationOfGoodsIdentificationTypeService,
```

Comment:
Consider giving this a more meaningful name when there are multiple service classes